fe/src/main/java/org/apache/doris/analysis/SelectStmt.java

```diff
@@ -1391,9 +1391,8 @@ private void rewriteSelectList(ExprRewriter rewriter) throws AnalysisException {
             CaseExpr caseExpr = (CaseExpr) item.getExpr();
 
             int childIdx = 0;
-            if (caseExpr.hasCaseExpr()
-                    && caseExpr.getChild(childIdx++).contains(Predicates.instanceOf(Subquery.class))) {
-                throw new AnalysisException("Only support subquery in binary predicate in case statement.");
+            if (caseExpr.hasCaseExpr()) {
+                childIdx++;
             }
             while (childIdx + 2 <= caseExpr.getChildren().size()) {
```

Comment:
maybe in 1397 line code , "<=" need to modify "<".
for example: case when expr0 then expr1 else expr2.   expr2 will be consider 'when expr' in In the following code

Reply:
after the  loop  the `childIdx` is 2  `childIdx + 2` is 4 not <= `caseExpr.getChildren().size()` (3) so expr2 will not be consider 'when expr' in In the following code